### ISSUE TITLE
Clearing events for bevy

### ISSUE THREAD

**Issue by Aceeri** [lucaspoffo/renet#31]
Might make more sense to do `.insert_resource(Events::<ServerEvent>::default());` and have custom clearing/updating systems set up, but this might be more excessively custom and more of a fault of the `Plugin` api not being super flexible.

**Comment by lucaspoffo** [comment]
What benefits could we have from having our clearing/updating of events instead of the default behavior?
Do you have any different behavior in mind?

**Comment by Aceeri** [comment]
Mostly just annoying in the case of having a fixed timestep for your networking, since the events will almost never show up in that case.

**Comment by Shatur** [comment]
You can run a system every frame that read events. Is there a use case to handle events with a fixed timestep too?

**Comment by Aceeri** [comment]
While yes there is a workaround, I'm just saying fixed timesteps for networking are very common usecases and it doesn't work with that.

**Comment by Shatur** [comment]
>  I'm just saying fixed timesteps for networking are very common usecases

But why you need to handle errors and receive messages with a fixed timestep? It doesn't make any sense.
You need only _process_ massages with a fixed timestep.

**Comment by Aceeri** [comment]
I'm not talking about the `receive_message` I'm talking about `ServerEvents` and `RenetError` which are already in that state of needing to be processed. I'd prefer it if clients connecting/disconnecting since this is what the processing code for server events usually looks like for me:
```rust
for event in server_events.iter() {
    match event {
        ServerEvent::ClientConnected(id, _) => {
            info!("player {} connected.", id);
            // Spawn player cube
            let player_entity = commands
                .spawn()
                .insert(Collider::capsule(Vec3::ZERO, Vec3::Y, 0.5))
                .insert(...);
        }
    }
}
```

**Comment by Shatur** [comment]
Still not sure why it better then handling disconnect right away?

**Comment by Aceeri** [comment]
The disconnect is already handled on renet's side, I just want to handle the cascading effects of that on gameplay in my timestep?

**Comment by Shatur** [comment]
Oh, got it. Yes, I agree with you.

**Comment by lucaspoffo** [comment]
Issue about this in bevy_rapier: https://github.com/dimforge/bevy_rapier/issues/216

**Comment by lucaspoffo** [comment]
Is there a way to clear events by default, and optionally let the user do it?

**Comment by Aceeri** [comment]
I think it might actually be better to not tie it to `update_system` since I think that would end up making the processing of messages/updates slower. But ya, I think I have a decent idea on how we could set this up to just be customizable, following bevy_rapier's example in how they let users set things up we could have some methods on `RenetPlugin` that give a `SystemSet` for "clear events" and then another for "update systems" and such.

**Comment by lucaspoffo** [comment]
Fixed by #34
